flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunner.java

```diff
@@ -367,12 +365,14 @@ public static TaskExecutor startTaskManager(
 			taskManagerServicesConfiguration.getSystemResourceMetricsProbingInterval());
 
 		TaskManagerServices taskManagerServices = TaskManagerServices.fromConfiguration(
+			configuration,
 			taskManagerServicesConfiguration,
 			taskManagerMetricGroup.f1,
 			resourceID,
 			rpcService.getExecutor(), // TODO replace this later with some dedicated executor for io.
 			EnvironmentInformation.getSizeOfFreeHeapMemoryWithDefrag(),
-			EnvironmentInformation.getMaxJvmHeapMemory());
+			EnvironmentInformation.getMaxJvmHeapMemory(),
+			localCommunicationOnly);
```

Comment:
Why did you change this? The intention was that `TaskManagerServicesConfiguration` contains all configuration values necessary to instantiate the `TaskManagerServices`.

Reply:
agreed, will put all configuration data into `TaskManagerServicesConfiguration`